### PR TITLE
Populate remaining event fields

### PR DESCRIPTION
**Pull request by spenes**
At the moment, some of the event fields aren't populated at all in the generated events. This PR introduces necessary changes to populate those fields as well.

### CODE REVIEW

File: core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/EventType.scala [hunk 1]
```diff
@@ -19,12 +19,13 @@ case class EventFrequencies(
   unstruct: Int,
   pageView: Int,
   pagePing: Int,
+  remainingFields: Int,
```
Comment (pondzix) [inline core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/EventType.scala:22]: If we add new field here, do we also need default [config](https://github.com/snowplow-incubator/snowplow-event-generator/blob/b9000d1f0e355684f8235eb1a3736615f7555dd5/sinks/src/main/resources/application.conf#L22) for it?

Comment (spenes) [reply core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/EventType.scala:22]: Yep, looks like it. Thanks for the heads up!

File: core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/EventType.scala [hunk 2]
```diff
@@ -45,6 +46,8 @@ object EventType {
 
   case object PagePing extends EventType
 
+  case object RemainingFields extends EventType
```
Comment (benjben) [inline core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/EventType.scala:49]: `RemainingFields` is not a type of event, like `Unstruct` or `PageView`

File: core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/RemainingFields.scala [hunk 3]
```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2021-2022 Snowplow Analytics Ltd. All rights reserved.
```
Comment (benjben) [inline core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/event/RemainingFields.scala:2]:
```suggestion
 * Copyright (c) 2023-present Snowplow Analytics Ltd. All rights reserved.
```

**Review by benjben — CHANGES_REQUESTED** [review]
This does not sound right to have "remaining fields". Remaining from what? The fields should be set directly when an actual event (e.g. page view, unstruct, etc.) is generated

File: core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/enrichment/Enrichment.scala [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.eventgen.protocol.enrichment
+
+trait Enrichment
```
Comment (benjben) [inline core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/enrichment/Enrichment.scala:15]: I see that each enrichment extends this trait but I don't see this type used anywhere else?

**Review by benjben — APPROVED** [review]
Thanks for the changes, it looks great, I love it that it's clear which enrichment populates which field :+1:

File: core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/enrichment/Enrichment.scala (hunk 4 above)
Comment (spenes) [reply core/src/main/scala/com/snowplowanalytics/snowplow/eventgen/protocol/enrichment/Enrichment.scala:15]: I started with some other design but change it to another during the implementation. It is remain from that, I forgot to remove it. I will remove it 👍

**Pull request merged**